packages/nx/src/project-graph/project-graph-builder.ts

```diff
@@ -128,7 +149,8 @@ export class ProjectGraphBuilder {
   addExplicitDependency(
     sourceProjectName: string,
     sourceProjectFile: string,
-    targetProjectName: string
+    targetProjectName: string,
+    type: DependencyType.static | DependencyType.dynamic = DependencyType.static
```

Comment:
This would be a breaking change.

Let's keep this API as is and deprecated it.

`addStaticDependency` and `addDynamicDependency` should be used instead.

Reply:
Ok, that resolves also the two above